src/win/util.c

```diff
@@ -1674,7 +1807,7 @@ int uv_os_gethostname(char* buffer, size_t* size) {
 
   uv__once_init(); /* Initialize winsock */
 
-  if (GetHostNameW(buf, UV_MAXHOSTNAMESIZE) != 0)
+  if (uv__gethostnamew_nt60(buf, UV_MAXHOSTNAMESIZE) != 0)
```

Comment:
@libuv/windows Is it worthwhile to use the Windows 8 function, when available, and only polyfill for <= 7:

```suggestion
  sGetHostNameW gethostnamew = pGetHostNameW == NULL ? uv__gethostnamew_nt60 : pGetHostNameW;
  if (gethostnamew(buf, UV_MAXHOSTNAMESIZE) != 0)
```

And in winapi.c
```
  ws2_32_module = LoadLibraryA("ws2_32.dll");
  if (ws2_32_module != NULL) {
    pGetHostNameW = (sGetHostNameW)
      GetProcAddress(ws2_32_module, "GetHostNameW");
```

Reply:
uv__gethostnamew_nt60 is basically a reverse engineered GetHostNameW to remove Win 8.1+ requirement.
They should behave exactly the same on Microsoft platforms.
GetProcAddress(ws2_32_module, "GetHostNameW"); will add an unnecessary branch in this case.
All 3 stages are also documented here:
https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-gethostname

However there can be some issues with alternative win32 implementations like wine.
GetHostNameW is just a wrapper for unix gethostname. It doesn't go through those stages defined in the documentation.
WSALookupService functions(stage 2) are not implemented, so my implementation will fallback to stage 3.

Reply:
Yes, wine is a reasonable case for it, as well as adopting any changes if Microsoft decides to change it